third_party/f2reduce/f2reduce.cpp

```diff
@@ -470,8 +470,8 @@ namespace f2reduce {
 
 void inplace_rref_strided(uint64_t *matrix, uint64_t rows, uint64_t cols, uint64_t stride) {
 
-    if (rows <= 1) {
-        // If the matrix has 0 or 1 rows, it must already be in RREF:
+    if (rows <= 1 || cols <= 1) {
+        // If the matrix has 0 or 1 rows or columns, it must already be in RREF:
```

Comment:
If the matrix has one column it may not be on RREF tho. This shoutcut should only be taken in the `cols ==  0` case

Reply:
I'm confused, how would cols == 0? Is it still a valid matrix?

Reply:
It's especially confusing if rows>=1 && cols == 0

Reply:
OK, then it would be better to also remove https://github.com/triton-lang/triton/blob/6f5baf6801b44e51b7ba8eedaa619e39c912bef6/lib/Tools/LinearLayout.cpp#L121 in this PR